### PR TITLE
Initial version

### PR DESCRIPTION
**Pull request by KlausVcb**
## Description

- A new library to allow data to be serizalized and deserialized both from and to nanoFramework and any .NETStandard 2.0 .Net flavor while providing stellar performance and the smallest possible payload.

## Motivation and Context

- A faster alternative to transferring json over the wire with smaller payloads

## How Has This Been Tested?

Test projects for both nanoFramework and .net8 are provided testing E2E as well as a payload > 65kB

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [x] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project (only if there are changes in source code).
- [x] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [x] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [x] I have added new tests to cover my changes.

## Release Notes

- **New Features**
	- Introduced two new GitHub Actions workflows: 
		- `pr-checks.yml` for automating pull request checks.
		- `update-dependencies.yml` for scheduled dependency updates.
	- Added a new section in `assets/readme.txt` with resources and links for community engagement.
	- New `version.json` file for automated versioning and release management.
	- New `packages.lock.json` files for `nanoFramework.Protobuf`, `nanoFramework.Protobuf.Test.nano`, `nanoFramework.Protobuf.NET`, and `nanoFramework.Protobuf.Test.NET` projects to manage dependencies effectively.

- **Improvements**
	- Updated `azure-pipelines.yml` to define a CI/CD pipeline with manual triggers and enhanced testing capabilities.
	- Introduced an `azure-bootstrap.yml` configuration for branch-based build triggers in Azure DevOps.

### CODE REVIEW

**Comment by KlausVcb** [comment]
First version up for review. Undoubtfully there is stuff missing or requires further updates :-D

File: .github/workflows/update-dependencies.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  repository_dispatch:
+    types: update-dependencies
+
+jobs:
+  update-dependencies:
+    name: nanoFramework
+    uses: nanoframework/nf-tools/.github/workflows/update-dependencies.yml@main
+    secrets: inherit
+    with:
+      solutionsToCheck: 'nanoFramework.Protobuf.sln'
```
[the remark this answers is not in the document]
Comment (josesimoes) [reply .github/workflows/update-dependencies.yml:21]: This is an internal maintained template. OK to use without versioning.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        inputs:
+          path: '$(System.DefaultWorkingDirectory)'
+          artifactName: 'Artifacts'
+
+- ${{ each appComponents in parameters.appComponents }}:   
+  - template: azure-pipelines-templates/device-test.yml@templates
+    parameters:
+      appComponents: ${{ appComponents }}
+      unitTestRunsettings: 
+        - 'nanoFramework.Protobuf.Test.Nano/nano.runsettings,nanoFramework.Protobuf.Test.Nano/bin/Release/NFUnitTest.dll'
```
[the remark this answers is not in the document]
Comment (josesimoes) [reply azure-pipelines.yml]: those are already part the run-settings config so no need for it

File: azure-bootstrap.yml [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+      else
+          echo "Failed to trigger pipeline. HTTP Status Code: $http_code"
+          echo "Response: $content"
+          exit 1
+      fi
+    displayName: 'Trigger Azure DevOps Pipeline'
+    env:        
+      BUILD_SOURCEBRANCH: $(Build.SourceBranch)
+      AZURE_DEVOPS_PAT: $(AZURE_DEVOPS_PAT)
+
```
[the remark this answers is not in the document]
Comment (josesimoes) [reply azure-bootstrap.yml:131]: @Ellerbach  care to review coderabitai suggestion?

File: azure-bootstrap.yml [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+              exit 1
+          fi
+
+          # Wait for a while before checking again
+          sleep 30
+      done
+    displayName: 'Monitoring Azure DevOps pipeline'
+    env:
+      run_id: $(run_id)
+      AZURE_DEVOPS_PAT: $(AZURE_DEVOPS_PAT)
```
[the remark this answers is not in the document]
Comment (josesimoes) [reply azure-bootstrap.yml:169]: @Ellerbach  care to review coderabitai suggestion?

File: azure-pipelines.yml [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+          status: 'failure'
+          webhookUrl: '$(DiscordWebhook)'
+          message: ''
+
+      - task: PublishPipelineArtifact@1
+        displayName: Publish Pipeline Artifact copy
+        inputs:
+          path: '$(System.DefaultWorkingDirectory)'
+          artifactName: 'Artifacts'
+
```
[the remark this answers is not in the document]
Comment (josesimoes) [reply azure-pipelines.yml:79]: @Ellerbach does it make sense to use `$(System.DefaultWorkingDirectory)/**/bin/$(buildConfiguration)` ? the packages dir, for sure it doesn't

**Review by josesimoes — APPROVED** [review]
Ready for release! 
Thank you for the contribution. 👌🏻 🔝

File: azure-pipelines.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (josesimoes) [reply azure-pipelines.yml:79]: @Ellerbach resolving this for the moment, but I think its worth looking into the suggestion.

File: azure-bootstrap.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (josesimoes) [reply azure-bootstrap.yml:169]: @Ellerbach resolving this for the moment, but I think its worth looking into the suggestion.

File: azure-bootstrap.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (josesimoes) [reply azure-bootstrap.yml:131]: @Ellerbach resolving this for the moment, but I think its worth looking into the suggestion.

**Pull request merged**